### PR TITLE
Syw UId2-1260 cstg distinguishing bits in tokens 

### PR DESCRIPTION
**Pull request by sunnywu**
- added 2 bit flags into UserIdentity.privacyBits - if token generated from cstg and if the cstg token is opted out.
- if we create a optout token in cstg call and when we request to refresh the produced refresh token, the bit flags will persist in the refreshed token's response, including the new ad tokens and new refresh token's UserIdentity objects

### CODE REVIEW

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java [hunk 1]
```diff
@@ -2566,6 +2584,15 @@ void cstgOptedOutTest(boolean optOutExpected, String id, IdentityType identityTy
...
 
+                    if(optOutExpected) {
+                        assertTrue(PrivacyBits.fromInt(advertisingToken.userIdentity.privacyBits).isClientSideTokenGenerated());
+                        assertTrue(PrivacyBits.fromInt(advertisingToken.userIdentity.privacyBits).isClientSideTokenOptedOut());
+                    }
+                    else {
+                        assertTrue(PrivacyBits.fromInt(advertisingToken.userIdentity.privacyBits).isClientSideTokenGenerated());
+                        assertFalse(PrivacyBits.fromInt(advertisingToken.userIdentity.privacyBits).isClientSideTokenOptedOut());
+                    }
+
```
Comment (mcollins-ttd) [inline src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java:2595]: suggestion:

```suggestion
                    assertTrue(PrivacyBits.fromInt(advertisingToken.userIdentity.privacyBits).isClientSideTokenGenerated());
                    assertEquals(optOutExpected, PrivacyBits.fromInt(advertisingToken.userIdentity.privacyBits).isClientSideTokenOptedOut());
```

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java [hunk 2]
```diff
@@ -2585,9 +2612,13 @@ else if(identityType == IdentityType.Phone) {
 
                     if(optOutExpected) {
                         assertArrayEquals(TokenUtils.getFirstLevelHashFromIdentityHash(getSha256(expectedOptedOutIdentity), firstLevelSalt), refreshToken.userIdentity.id);
+                        assertTrue(PrivacyBits.fromInt(refreshToken.userIdentity.privacyBits).isClientSideTokenGenerated());
```
Comment (mcollins-ttd) [inline src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java:2615]: nitpick: This is the same in both branches, can we move it outside the if-else?

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java [hunk 3]
```diff
@@ -2614,7 +2645,72 @@ else if(identityType == IdentityType.Phone) {
...
+                        assertTrue(PrivacyBits.fromInt(adTokenFromRefresh.userIdentity.privacyBits).isClientSideTokenGenerated());
+                        assertTrue(PrivacyBits.fromInt(refreshTokenAfterRefresh.userIdentity.privacyBits).isClientSideTokenGenerated());
+                        if(optOutExpected) {
+                            assertTrue(PrivacyBits.fromInt(adTokenFromRefresh.userIdentity.privacyBits).isClientSideTokenOptedOut());
+                            assertTrue(PrivacyBits.fromInt(refreshTokenAfterRefresh.userIdentity.privacyBits).isClientSideTokenOptedOut());
+                        }
+                        else {
+                            assertFalse(PrivacyBits.fromInt(adTokenFromRefresh.userIdentity.privacyBits).isClientSideTokenOptedOut());
+                            assertFalse(PrivacyBits.fromInt(refreshTokenAfterRefresh.userIdentity.privacyBits).isClientSideTokenOptedOut());
+                        }
```
Comment (mcollins-ttd) [inline src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java:2711]: suggestion:

```suggestion
                        assertEquals(optOutExpected, PrivacyBits.fromInt(adTokenFromRefresh.userIdentity.privacyBits).isClientSideTokenOptedOut());
                        assertEquals(optOutExpected, PrivacyBits.fromInt(refreshTokenAfterRefresh.userIdentity.privacyBits).isClientSideTokenOptedOut());
```

**Review by mcollins-ttd — APPROVED** [review]
Approved, left some suggestions.

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java (hunk 2 above)
Comment (sunnywu) [reply src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java]: fixed

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java (hunk 1 above)
Comment (sunnywu) [reply src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java]: fixed as suggested

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java (hunk 3 above)
Comment (sunnywu) [reply src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java]: fixed as suggested

**Comment by sunnywu** [comment]
Thanks @mcollins-ttd. will leave this PR open until i merge my other PRs back to cstg branch

**Pull request merged**